### PR TITLE
feat: add `mkTempDir` to `FileWrite` and `FileWriteWithResult`

### LINKED ISSUE #9226: Add `mkTemporaryDirectory` to File AEs

### PR DESCRIPTION
**Pull request by sockmaster27**
Fixes #9226

### CODE REVIEW

**Comment by magnus-madsen** [comment]
What's the status on this?

**Comment by sockmaster27** [comment]
I'm stumped on why the `FileAlreadyExists` exception isn't caught in the handler

**Comment by magnus-madsen** [comment]
Can you be more precise / point to where you mean?

File: main/src/library/FileWrite.flix [hunk 1]
```diff
@@ -100,9 +107,9 @@ mod FileWrite {
     ///
     /// Handles the `FileWrite` effect of the given function `f`.
     ///
-    /// In other words, re-interprets the `FileWrite` effect using the `IO` effect.
+    /// In other words, re-interprets the `FileWrite` effect using the `Sys` and `IO` effects.
     ///
-    pub def handle(f: a -> b \ ef): a -> Result[IoError, b] \ (ef - FileWrite) + IO = x ->
+    pub def handle(f: a -> b \ ef): a -> Result[IoError, b] \ (ef - FileWrite) + {Sys, IO} = x ->
```
Comment (magnus-madsen) [inline main/src/library/FileWrite.flix:112]: I would like to avoid the Sys effect here. So lets just call `getTemporaryDirectory` directly from Java.

**Comment by magnus-madsen** [comment]
Test failures...

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**